### PR TITLE
feat(wasm-dpp): DashPlatformProtocol tests

### PR DESCRIPTION
**Pull request by antouhou**
## Issue being fixed or feature implemented

Missing tests for the main DPP class

## What was done?

Add tests for the main DPP class, and refactored it to allow for version/stateRespository getters and setters to work correctly

## How Has This Been Tested?

Run tests

## Breaking Changes

None

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

**Comment by markin-io** [comment]
And clippy :)

File: packages/wasm-dpp/src/dash_platform_protocol.rs [hunk 1]
```diff
@@ -97,6 +88,102 @@ impl DashPlatformProtocol {
     pub fn state_transition(&self) -> StateTransitionFacadeWasm {
         self.state_transition.clone()
     }
+
+    #[wasm_bindgen(getter = protocolVersion)]
```
[the remark this answers is not in the document]
Comment (antouhou) [reply packages/wasm-dpp/src/dash_platform_protocol.rs:92]: It was a getter in the original test, but I can make a function as well :)

File: packages/wasm-dpp/test/unit/DashPlatformProtocol.spec.js [hunk 2]
```diff
@@ -1,39 +1,27 @@
-const { default: Ajv } = require('ajv/dist/2020');
-
 const protocolVersion = require('@dashevo/dpp/lib/version/protocolVersion');
 
-const DashPlatformProtocol = require('@dashevo/dpp/lib/DashPlatformProtocol');
-const JsonSchemaValidator = require('@dashevo/dpp/lib/validation/JsonSchemaValidator');
-
-const createStateRepositoryMock = require('@dashevo/dpp/lib/test/mocks/createStateRepositoryMock');
+// const DashPlatformProtocol = require('@dashevo/dpp/lib/DashPlatformProtocol');
```
[the remark this answers is not in the document]
Comment (antouhou) [reply packages/wasm-dpp/test/unit/DashPlatformProtocol.spec.js]: Done 👍

File: packages/wasm-dpp/src/dash_platform_protocol.rs [hunk 3]
```diff
@@ -34,47 +38,34 @@ pub struct DPPOptions {
...
-            protocol_version,
-            LATEST_VERSION,
-            COMPATIBILITY_MAP.clone(),
-        ));
-        let public_keys_validator = Arc::new(PublicKeysValidator::new(bls).unwrap());
-
-        let identity_facade =
-            IdentityFacadeWasm::new(protocol_version_validator.clone(), public_keys_validator);
+        // // TODO: wrap whole thing around rs-dpp/dash_platform_protocol?
+        // let options: DPPOptions = with_js_error!(serde_wasm_bindgen::from_value(options))?;
```
[the remark this answers is not in the document]
Comment (antouhou) [reply packages/wasm-dpp/src/dash_platform_protocol.rs]: Done 👍

File: packages/wasm-dpp/src/dash_platform_protocol.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (antouhou) [reply packages/wasm-dpp/src/dash_platform_protocol.rs:89]: Added an additional getter function

**Review by markin-io — DISMISSED** [review]

**Pull request merged**